test/libsolidity/semanticTests/various/eof/create_calldata.sol

```diff
@@ -0,0 +1,19 @@
+contract C {
+	bytes public s;
+	constructor(uint256 x) {
+		// Due to a bug in EVMHost, msg.data used to contain initcode and constructor arguments.
+		s = msg.data;
+		assert(msg.data.length == 32);
```

Comment:
I'd also check that the values match.

```suggestion
		// On EOF msg.data contains constructor arguments, while on legacy it's empty
		// (arguments are appended to the code).
		s = msg.data;
		assert(msg.data.length == 32);
		y = abi.decode(msg.data, (uint));
		assert(x == y);
```

Also would not hurt to have more than one argument.

Reply:
done.